sdk/src/types/block/protocol.rs

```diff
@@ -156,7 +156,7 @@ pub mod dto {
         #[cfg_attr(feature = "serde", serde(rename = "version"))]
         pub protocol_version: u8,
         pub network_name: String,
-        pub bech32_hrp: String,
+        pub bech32_hrp: Hrp,
```

Comment:
This breaks the wallet `Error: Json(Error("invalid type: map, expected a string representing the value", line: 1, column: 818))` because it's used in the client options that are stored in the db

Reply:
`Hrp` is `FromStr` so maybe just applying `string_serde_impl!` to it will solve this issue?

Reply:
So do we need to merge the migration PR first, and then add a migration in this PR?

Reply:
I think so, if we don't want to add another workaround before